### ISSUE TITLE
Enhance type-checking

### ISSUE THREAD

**Issue by jafri** [uuosio/ascdk#10]
Currently a few type-checking errors pop up:

```
Cannot find name 'contract'.ts(2304)
```

This could be fixed by exporting stub functions for decorators like:

```
function contract(_: string): any {}
function action(_: string): any {}
function table(_: string): any {}
```

Another type-checking error that pops up while developing is:
```
Type 'AccountsTable' does not satisfy the constraint 'MultiIndexValue'.ts(2344)
```

This is because the rest of the fields are only added as part of the extension after processing, perhaps the other fields for MultiIndexValue should become optional

**Comment by learnforpractice** [comment]
Yes, you are right. Help me fix it if you can.

**Comment by jafri** [comment]
@learnforpractice do you know how to make `Cannot find name 'u64'` type errors go away while developing for AS?

**Comment by learnforpractice** [comment]
Nope.

**Comment by learnforpractice** [comment]
I found an AssemblyScript plugin for VSCode:

https://marketplace.visualstudio.com/items?itemName=saulecabrera.asls

**Comment by learnforpractice** [comment]
> I found an AssemblyScript plugin for VSCode:
> 
> https://marketplace.visualstudio.com/items?itemName=saulecabrera.asls

This plugin is not working on Mac OSX platform due to the following issue:

https://github.com/Shopify/vscode-as/issues/20

**Comment by jafri** [comment]
@learnforpractice is this right?

<img width="772" alt="Screen Shot 2022-02-20 at 10 20 16 AM" src="https://user-images.githubusercontent.com/13677357/154855381-ca352419-a9ec-4a4b-9688-3f491408fb5e.png">

**Comment by jafri** [comment]
The main type issues at the moment are `MultiIndexValue`, `Packer`, and `.new` (on MultiIndexValue through Tpl)

**Comment by learnforpractice** [comment]
> 

Should be fixed by 4813106a913009757bf72993c899d9340f08fe5a
